shared/common-adapters/icon.constants.js

```diff
@@ -1,265 +1,1251 @@
-  'icon-phone-bw-revoke-64': true,
-  'icon-phone-bw-success-48': true,
-  'icon-phone-colors-24': true,
-  'icon-phone-colors-32': true,
-  'icon-phone-colors-48': true,
-  'icon-phone-colors-64': true,
-  'icon-phone-dark-blue-32': true,
-  'icon-phone-qr-code-32': true,
-  'icon-phone-qr-code-48': true,
-  'icon-phone-text-code-32': true,
-  'icon-phone-text-code-48': true,
-  'icon-placeholder-avatar-112-x-112': true,
-  'icon-placeholder-avatar-176-x-176': true,
-  'icon-placeholder-avatar-24-x-24': true,
-  'icon-placeholder-avatar-32-x-32': true,
-  'icon-placeholder-avatar-48-x-48': true,
-  'icon-placeholder-avatar-64-x-64': true,
-  'icon-placeholder-avatar-80-x-80': true,
-  'icon-progress-grey-animated': true,
-  'icon-progress-grey-static': true,
-  'icon-progress-white-animated': true,
-  'icon-progress-white-static': true,
-  'icon-reddit-logo-16': true,
-  'icon-reddit-logo-24': true,
-  'icon-reddit-logo-32': true,
-  'icon-reddit-logo-48': true,
-  'icon-reddit-logo-64': true,
-  'icon-subnav-folders-all': true,
-  'icon-subnav-folders-private': true,
-  'icon-subnav-folders-public': true,
-  'icon-terminal-32': true,
-  'icon-terminal-48': true,
-  'icon-twitter-logo-16': true,
-  'icon-twitter-logo-24': true,
-  'icon-twitter-logo-32': true,
-  'icon-twitter-logo-48': true,
-  'icon-twitter-logo-64': true,
-  'icon-update-64': true,
-  'icon-update-pause-64': true,
-  'icon-website-16': true,
-  'icon-website-24': true,
-  'icon-website-32': true,
-  'icon-website-48': true,
-  'icon-website-64': true}
-
-export type IconsType = $Enum<typeof Icons>
+type IconMeta = {
+  isFont: boolean,
+  gridSize?: number,
+  extension?: string,
+  charCode?: number,
+  require?: any,
+}
 
-export const fontSizes: {[key: string]: number} = {
-  'iconfont-caret-down': 8,
-  'iconfont-caret-right': 8,
-  'iconfont-identity-twitter': 16,
-  'iconfont-identity-github': 16,
-  'iconfont-identity-reddit': 16,
-  'iconfont-identity-website': 16,
-  'iconfont-identity-hn': 16,
-  'iconfont-coinbase': 16,
-  'iconfont-identity-bitcoin': 16,
-  'iconfont-identity-devices': 16,
-  'iconfont-identity-pgp': 16,
-  'iconfont-folder-private': 16,
-  'iconfont-folder-private-has-files': 16,
-  'iconfont-folder-public-has-files': 16,
-  'iconfont-folder-public': 16,
-  'iconfont-close': 16,
-  'iconfont-check': 16,
-  'iconfont-back': 16,
-  'iconfont-thunderbolt': 16,
-  'iconfont-remove': 16,
-  'iconfont-import': 16,
-  'iconfont-add': 16,
-  'iconfont-link': 16,
-  'iconfont-ellipsis': 16,
-  'iconfont-edit': 16,
-  'iconfont-tweet': 16,
-  'iconfont-open-browser': 16,
-  'iconfont-lock': 16,
-  'iconfont-proof-good': 24,
-  'iconfont-proof-broken': 24,
-  'iconfont-proof-pending': 24,
-  'iconfont-hamburger': 24,
-  'iconfont-people': 24,
-  'iconfont-folder': 24,
-  'iconfont-device': 24,
-  'iconfont-settings': 24,
-  'iconfont-search': 24,
-  'iconfont-chat': 24,
+export const iconMeta: {[key: string]: IconMeta} = {
```

Comment:
I think it would be useful to have this be `{[key: IconType]: IconMeta}` because we are always doing `iconMeta['foo'].{isFont,require...}` and it would be nice if flow could tell us when `foo` isn't an icon.

That'll probably involve making an explicit `IconType`. Then doing  `{[key: IconType]: IconMeta}`